nlmod/cache.py

```diff
@@ -112,6 +131,15 @@ def decorator(*args, cachedir=None, cachename=None, **kwargs):
                     )
                 dataset = func_args_dic.pop(key)
 
+        dataset = ds_contains(
```

Comment:
I think this raises an error when the dataset is None. Can be easily solved with a check first.